models/DataObject/ClassDefinition/Data/Hotspotimage.php

```diff
@@ -330,7 +330,10 @@ public function getDataFromEditmode($data, $object = null, $params = [])
             $data['hotspots'] = $rewritePath($data['hotspots']);
         }
 
-        return new DataObject\Data\Hotspotimage($data['id'] ?? null, $data['hotspots'] ?? [], $data['marker'] ?? [], $data['crop'] ?? []);
+        if ($data && isset($data['id']) && (int)$data['id'] > 0) {
```

Comment:
```suggestion
        if (isset($data['id']) && (int)$data['id'] > 0) {
```
should be enough

Reply:
merging anyways for release purpose, since this suggested change is not a big deal